demo/admin/src/App.tsx

```diff
@@ -149,3 +152,32 @@ class App extends React.Component {
     }
 }
 export default App;
+
+declare module "@comet/cms-admin" {
+    interface SitePreviewParams {
+        userGroup: GQLUserGroup;
+    }
+}
+
+function CustomSitePreview(props: RouteComponentProps) {
+    const [previewUserGroup, setPreviewUserGroup] = useSearchState("userGroup", (param) => param ?? "All");
+
+    return (
+        <SitePreview
+            {...props}
+            actions={
+                <Select
+                    value={previewUserGroup}
+                    onChange={(event) => {
+                        setPreviewUserGroup(event.target.value);
+                    }}
+                >
+                    <MenuItem value="All">All</MenuItem>
+                    <MenuItem value="Admin">Admin</MenuItem>
+                    <MenuItem value="User">User</MenuItem>
+                </Select>
+            }
+            additionalSitePreviewParams={{ userGroup: previewUserGroup as GQLUserGroup }}
```

Comment:
Alternatively, we could implement a `SitePreviewApi` that allows changing site preview params.